crates/rspack_core/src/options/compiler_options.rs

```diff
@@ -15,7 +17,7 @@ pub struct CompilerOptions {
   pub resolve: Resolve,
   pub resolve_loader: Resolve,
   pub module: ModuleOptions,
-  pub devtool: Devtool,
+  pub devtool: Mutex<Devtool>,
```

Comment:
The Mutex is not necessary at all, this should be readonly, all options are finalized at js side

Reply:
The `devtool` setting currently is not yet finalized on the JavaScript side. The `SourceMapDevToolPlugin` will configure it completely.

In Webpack, the [SourceMapDevToolModuleOptionsPlugin](https://github.com/webpack/webpack/blob/main/lib/SourceMapDevToolModuleOptionsPlugin.js) is used to set `useSourceMap` or `useSimpleSourceMap` to true, which signals that each module should generate a source map.

In rspack, I've used the `devtool` setting to achieve the aim .

Reply:
It's worth mentioning that a comment in the webpack code suggests the `useSourceMap` and `useSimpleSourceMap`  might be refactored in webpack 6, which is something I'm not very familiar with. You can see the comment here: https://github.com/webpack/webpack/blob/main/lib/Module.js#L169

Reply:
it's only a refactor without any logic change, refactor `useSourceMap` and `useSimpleSourceMap` to one enum

Reply:
// TODO refactor this -> options object filled from Factory

Reply:
ohh, then we can delay SourceMapDevToolModuleOptionsPlugin, but devtool shouldn't wrapped by Mutex

Reply:
If we don't wrap `devtool` with a Mutex, what method should I use to modify it in `SourceMapDevToolModulePlugin`?

Reply:
@ahabhgk I have used `RwLock` instead of `Mutex`. It's enough for the goal of modifying `devtool` in `compilation` hook and it's more cheap.

Reply:
Do you have any other methods?